internal/tools/k8s/helpers.go

```diff
@@ -12,14 +14,11 @@ import (
 // WaitForPodRunning will poll a given kubernetes pod at a regular interval for
 // it to enter the 'Running' state. If the pod fails to become ready before
 // the provided timeout then an error will be returned.
-func (kc *KubeClient) WaitForPodRunning(namespace, podName string, timeout int) (*corev1.Pod, error) {
-	timer := time.NewTimer(time.Duration(timeout) * time.Second)
-	defer timer.Stop()
-
+func (kc *KubeClient) WaitForPodRunning(namespace, podName string, ctx context.Context) (*corev1.Pod, error) {
 	for {
 		select {
-		case <-timer.C:
-			return &corev1.Pod{}, errors.New("timed out waiting for pod to become ready")
+		case <-ctx.Done():
+			return &corev1.Pod{}, errors.Wrap(ctx.Err(), "timed out waiting for pod to become ready")
```

Comment:
return `nil` here instead of an empty struct?

Reply:
Can't comment on the `time.Sleep` below (I'm reviewing from a specific commit), but might we enable cancelling that timer (so we don't have to wait up to 5 seconds after a cancelled context) by structuring the code as follows instead?

```go
for {
    pod, err := ...

    select {
        case <-ctx.Done():
            return nil, errors.Wrap(...)
        case <-time.After(5 * time.Second):
    }
}
```

Reply:
If we do that, we will then have to wait for 5 seconds even if the pod is immediately ready.

Reply:
I suppose we could do a pod check out of the loop and then enter the loop if that doesn't succeed...

Hmm, not sure if that behavior is desirable or not.

Reply:
Hmm, I’m not sure I follow — why wouldn’t we return after the `pod, err` line using the same semantics as the existing code?

Reply:
Oh, I see what you are proposing now! I will push up something in a bit.